Unify description of reverse domain name notation, strong recommendation to use it for variable and terminal kinds (#1977)
modelica/fmi-standard: docs/2_4_common_schema.adoc

Each `Annotation` element contains a required `type` attribute, which contains the namespace for that annotation.
The content of the `Annotation` element can be arbitrary XML data, and can make use of XML namespaces and XML schemas for combined validation where appropriate.

The namespace mechanism for the `type` attribute is based on <<reverse-DNS,reverse domain name notation>>:
The originator of a specification for additional data specifies a domain name under their control as the namespace for the additional data, in order to avoid conflicts due to name collisions.

> For example this such standardized annotations could be defined by a <<LS-reverse-DNS,layered standard>>.

Reviewer: ```suggestion
Such standardized annotations can be defined by a <<LS-reverse-DNS,layered standard>>, for example.
```

The normalized string attribute `matchingRule` describes the rules for variable matching in a connection of terminals.
As detailed in <<table-predefined-matching-rules>>, there are three predefined matching rules: `plug`, `bus`, and `sequence`.
Other standards may define new matching rules.
In order to avoid ambiguities and conflicts, rule names must follow the <<reverse-dns,reverse domain name notation>> of a domain that is controlled by the entity defining the semantics and content of the additional entries.

> Such a rule definition could be part of a <<LS-reverse-DNS>>layered standard>>.

Reviewer: ```suggestion
Such rule definitions can be part of a <<LS-reverse-DNS>>layered standard>>, for example.
```
